src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -99,6 +99,34 @@ uint8_t MapToMatterWiFiVersionCode(wifi_link_mode wifiVersion)
     return (static_cast<uint8_t>(wifiVersion) - 1);
 }
 
+// Matter expectations towards Wi-Fi security type codes are unaligned with
+// what wpa_supplicant provides. This function maps supplicant codes
+// to the ones defined in the Matter spec (11.14.3.2. SecurityType enum)
+uint8_t MapToMatterSecurityType(wifi_security_type securityType)
+{
+    if (securityType < WIFI_SECURITY_TYPE_NONE || securityType > WIFI_SECURITY_TYPE_SAE ||
+        WIFI_SECURITY_TYPE_UNKNOWN == securityType)
+    {
+        ChipLogError(DeviceLayer, "Unspecified security type detected");
+        return EMBER_ZCL_SECURITY_TYPE_UNSPECIFIED;
+    }
+
+    switch (securityType)
+    {
+    case WIFI_SECURITY_TYPE_NONE:
+        return EMBER_ZCL_SECURITY_TYPE_NONE;
+    case WIFI_SECURITY_TYPE_PSK:
+    case WIFI_SECURITY_TYPE_PSK_SHA256:
+        return EMBER_ZCL_SECURITY_TYPE_WPA2;
+    case WIFI_SECURITY_TYPE_SAE:
+        return EMBER_ZCL_SECURITY_TYPE_WPA3;
+    default:
+        break;
+    }
+
+    return (static_cast<uint8_t>(securityType) - 1);
```

Comment:
What is this supposed to do?

Reply:
It's not needed here, I will replace it with EMBER_ZCL_SECURITY_TYPE_UNSPECIFIED.

Reply:
This is the default mapping for the WiFi version enumeration based on the spec. I'm not sure if this applies to the security mode as well.